internal/healthchecks/api_check.go

```diff
@@ -97,17 +98,21 @@ func createMonitoringPingRequest(gceMetadata resourcedetector.GCEResource) *moni
 // time series point with empty values to an Ops Agent specific metric.
 // This method mirrors the "(c *Client) Ping" method in "cloud.google.com/go/logging".
 func monitoringPing(ctx context.Context, client monitoring.MetricClient, gceMetadata resourcedetector.GCEResource) error {
+	// The retry logic fixes b/291631906 when the `CreateTimeSeries` request is retried very quickly
+	// resulting in an `InvalidArgument` error due a maximum write rate of one point every 5 seconds.
+	// https://cloud.google.com/monitoring/quotas 
+	pingBackoff := backoff.WithMaxRetries(backoff.NewConstantBackOff(6 * time.Second), MaxMonitoringPingRetries)
+	ticker := backoff.NewTicker(pingBackoff)
```

Comment:
`backoff.Retry` is probably simpler than using a ticker + loop. Example: https://github.com/GoogleCloudPlatform/ops-agent/blob/ef7383bce689b86bd28596a9e37400877b6fcb86/integration_test/gce/gce_testing.go#L957-L961

Reply:
This is my thought process for not using `backoff.Retry` and going for the `ticker + loop` instead.

1. Not every `error` returned from the `client.CreateTimeSeries(ctx, createMonitoringPingRequest(gceMetadata))` operation implies the need for a retry, only `InvalidArgument` needs a retry. All other errors are helpful to determine the API config state.
2. We would need to use `PermanentError` to determine `good` and `bad` (retriable) errors.
3. There would be a need for `Unwrap` logic for the `good` errors.
4. `ticker + loop` is simpler to read and understand.

Reply:
Well, took a crack at it and it looks nicer with `backoff.Retry` since there is no need to `Unwrap` the errors because `runMonitoringCheck` uses `As` to verify the error chain.